In [2]:
import pandas as pd
from pathlib import Path

data_path = Path("../data/child_mom_scored.csv")
df = pd.read_csv(data_path)

df.head()
df[["rf_proba", "risk_band"]].describe(include="all")
df["risk_band"].value_counts(normalize=True)


risk_band
low       0.585260
high      0.390579
medium    0.024161
Name: proportion, dtype: float64

Sanity check of scored data

We reloaded `data/child_mom_scored.csv` to verify that the scoring script
created the same distributions as in the policy/simulation notebook.

- The `rf_proba` and `risk_band` columns are present for all 5,753 records.
- Risk-band split:
  - Low risk: ~58.5% of children
  - Medium risk: ~2.4%
  - High risk: ~39.1%
- These proportions match the earlier analysis in `03_policy_and_simulation.ipynb`,
  confirming that the saved model (`models/rf_missed_any.joblib`) and scoring
  pipeline are working as expected.
